src/index.js

```diff
@@ -148,35 +149,20 @@ const fetch = async (root, selectedMode, sidekey, callback) => {
                     callback(payload)
                 }
             }
-        } while(!!hasMessage)
+        } while(!!hasMessage && messages.length < localLimit)
         return { messages, nextRoot }
     } catch (e) {
-        return console.error('failed to parse: ', e)
+        console.error('failed to parse: ', e)
         return e
     }
 }
 
-const fetchSingle = async (root, mode, sidekey, rounds = 81) => {
-    let address = root
-    if (mode === 'private' || mode === 'restricted') {
-        address = hash(root, rounds)
-    }
-    const { findTransactions } = composeAPI({ provider })
-    const hashes = await findTransactions({
-        addresses: [address]
-    })
-
-    const messagesGen = await txHashesToMessages(hashes)
-    for (let message of messagesGen) {
-        try {
-            // Unmask the message
-            const { payload, next_root } = decode(message, sidekey, root)
-            // Return payload
-            return { payload, nextRoot: next_root }
-        } catch (e) {
-            console.error('failed to parse: ', e)
-        }
-    }
+const fetchSingle = async (root, selectedMode, sidekey) => {
+    const response = await fetch(root, selectedMode, sidekey, undefined, 1)
```

Comment:
Is there a specific reason to not explicitly set the `callback` to `null`? 

same is on line 163.

Reply:
Not sure I understand your question, we don't have a callback passed to fetchSingle we just return the value, so we pass undefined to fetch to not use the callback mechanism.